docs/migration-service.md

```diff
@@ -122,96 +201,288 @@ flowchart TD
+> 1. Computes the launcher compose hash by filling the template with the Docker image hash
+> 2. Adds both the Docker image hash and launcher compose hash to their respective allowed lists
+>
+> This deterministic derivation ensures the launcher configuration always matches the voted Docker image, eliminating the need for separate voting. The same pattern is used for MPC nodes with `vote_code_hash()`.
+
+#### Migration Methods
+
+The contract provides the following methods:
+
+- **`start_node_migration(destination_node_info: ParticipantInfo)`** - Initiates a node migration:
+    - Called by the node operator
+    - Creates an `OngoingNodeMigration` record for the given `AccountId`
+    - Stores the destination node's `ParticipantInfo` (new TLS keys, etc.)
+    - Can be called multiple times to update the destination node info (only the last value is retained)
+    - Returns an error if the protocol is not in `Running` state
+    - Returns an error if caller is not a current participant
+
+- **`cancel_node_migration()`** - Cancels an ongoing node migration:
+    - Called by the node operator
+    - Removes the `OngoingNodeMigration` record for the given `AccountId`
+    - Useful if the new node is not functioning correctly or wrong information was provided
+
+- **`conclude_node_migration(keyset: &Keyset)`** - Finalizes a node migration:
+    - Called by the new node after receiving keyshares from backup service
+    - Verifies the provided `keyset` matches the expected key event IDs for this epoch
+    - Replaces the old node's `ParticipantInfo` with the new node's info in the current participant set
+    - Removes the `OngoingNodeMigration` record
+    - Returns an error if the protocol is not in `Running` state
+    - Returns an error if no ongoing migration exists for the caller
+
+- **`register_backup_service(backup_service_info: BackupServiceInfo)`** - Registers or updates backup service:
+    - Called by the node operator
+    - Stores the backup service's public key and URL for the node operator's account
+    - Defines or overrides the `BackupServiceInfo` for the node operator
+    - Can be called in any protocol state (`Running`, `Initializing`, or `Resharing`)
+    - Returns an error if caller is not a current participant
+
+> **Hard Launch Extension (Planned):** For hard launch, `register_backup_service()` will require an `attestation` and `operator_account_pk` parameter. The contract will verify the attestation validity, Docker image hash, and that the `ReportData` includes both the TLS public key and operator's account public key (`SHA3-384(tls_public_key || operator_account_pk)`). This cryptographically binds the backup service TEE to the specific operator, preventing a malicious backup service from registering under a different operator's account. Backup services will need to refresh attestations before expiration.
```

Comment:
It would be nice to have a code block and agree (more or less) on the type / interface we want here.